hub/core/storage/provider.py

```diff
@@ -121,7 +124,25 @@ def __len__(self):
             int: the number of files present inside the root.
         """
 
+    def enable_readonly(self):
+        """Enables read-only mode for the provider."""
+        self.read_only = True
+
+    def disable_readonly(self):
+        """Disables read-only mode for the provider."""
+        self.read_only = False
+
+    def assert_readonly(self):
```

Comment:
nit-pick `assert_readonly` tells me it's a test function, maybe `check_readonly` or `except_readonly`? not sure of good naming but `assert` maybe not the best

Reply:
Lol I thought someone might bring this up. Good call.